modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -175,37 +184,72 @@ public RabbitMQContainer withSSL(
         final MountableFile caFile,
-        final SslVerification verify,
-        boolean failIfNoCert
+        final SslVerification verify
     ) {
-        return withSSL(keyFile, certFile, caFile, verify)
-            .withEnv("RABBITMQ_SSL_FAIL_IF_NO_PEER_CERT", String.valueOf(failIfNoCert));
+        return withSSL(keyFile, certFile, caFile, verify, true);
     }
 
+    @SneakyThrows(IOException.class)
     public RabbitMQContainer withSSL(
         final MountableFile keyFile,
         final MountableFile certFile,
         final MountableFile caFile,
-        final SslVerification verify
+        final SslVerification verify,
+        boolean failIfNoCert,
+        int verificationDepth
     ) {
-        return withEnv("RABBITMQ_SSL_CACERTFILE", "/etc/rabbitmq/ca_cert.pem")
-            .withEnv("RABBITMQ_SSL_CERTFILE", "/etc/rabbitmq/rabbitmq_cert.pem")
-            .withEnv("RABBITMQ_SSL_KEYFILE", "/etc/rabbitmq/rabbitmq_key.pem")
-            .withEnv("RABBITMQ_SSL_VERIFY", verify.value)
-            .withCopyFileToContainer(certFile, "/etc/rabbitmq/rabbitmq_cert.pem")
+        Path advancedConfig = File.createTempFile("advanced-custom", ".config").toPath();
+        String advancedContent = createAdvancedConfigFileContent(
+            "/etc/rabbitmq/ca_cert.pem",
+            "/etc/rabbitmq/rabbitmq_cert.pem",
+            "/etc/rabbitmq/rabbitmq_key.pem",
+            verify,
+            failIfNoCert,
+            verificationDepth
+        );
+        Files.write(advancedConfig, advancedContent.getBytes(StandardCharsets.UTF_8));
+        return withCopyFileToContainer(certFile, "/etc/rabbitmq/rabbitmq_cert.pem")
             .withCopyFileToContainer(caFile, "/etc/rabbitmq/ca_cert.pem")
-            .withCopyFileToContainer(keyFile, "/etc/rabbitmq/rabbitmq_key.pem");
+            .withCopyFileToContainer(keyFile, "/etc/rabbitmq/rabbitmq_key.pem")
+            .withEnv("RABBITMQ_ADVANCED_CONFIG_FILE", "/etc/rabbitmq/advanced-custom.config")
+            .withCopyFileToContainer(MountableFile.forHostPath(advancedConfig), "/etc/rabbitmq/advanced-custom.config");
+    }
+
+    private String createAdvancedConfigFileContent(
+        String caCertFilePath,
+        String certFilePath,
+        String keyFilePath,
+        final SslVerification verify,
+        boolean failIfNoCert,
+        int verificationDepth
+    ) {
+        return String.format(
+            "[\n" +
+            "  {rabbit, [{ssl_listeners, [%d]},\n" +
+            "            {ssl_options,   [{cacertfile,           \"%s\"},\n" +
+            "                             {certfile,             \"%s\"},\n" +
+            "                             {keyfile,              \"%s\"},\n" +
+            "                             {verify,               %s},\n" +
+            "                             {fail_if_no_peer_cert, %b},\n" +
+            "                             {depth,                %d}]}]}\n" +
+            "].",
```

Comment:
I think we can use the `conf` format instead of `config`

Reply:
If we define an advanced configuration file, we have no choice to use the Erlang format https://www.rabbitmq.com/configure.html#advanced-config-file

Reply:
suggestions are welcome here. I'm not a rabbitmq expert but I did my comments after reviewing the docs so probably missing something.